Release v3.0

sjoerd-bouma opened the pull request:
After well over 2000 commits... perhaps it's time to make a new release, with lots of exciting new features (& probably new bugs!).

- [x] Merge #790 ?
- [x] Update changelog
- [x] Update version number

Probably for a future release (?):

- [ ] #396
- [x] #823

fschlueter commented:
Can we fix the compatibility test of 3.7 by fixing the nuradio scipy dependency to >1.8? Would that have unwanted side effects?

fschlueter committed:
Merge branch 'develop' into update-rnog-simulation-example

fschlueter committed:
Add back noise.py example script which was removed in a prev commit but since PR #841 is already merged should not be removed by this PR

fschlueter committed:
Improve calculate_vrms_from_temperature. Use it in noise.py and simulation.py.

fschlueter committed:
Update get_signal_to_noise_ratio

martin01021207 committed:
Update trace_utilities.py

MijnheerD committed:
Add module to generate and add galactic noise to efields

MijnheerD committed:
Updated changelog.txt

fschlueter committed:
Merge branch 'add_rnog_data_processing_script' into add_rnog_extra_variables

martin01021207 committed:
Update channelSignalReconstructor.py

Fixed the issues from conflicts:
Corrected functions' names that had already been changed

martin01021207 committed:
Merge pull request #849 from nu-radio/add_rnog_extra_variables

Adding RNO-G Extra Variables and Moving Analysis Variable Calculating Functions Around

fschlueter committed:
Extend PR template

MijnheerD committed:
Add function to calculate efield amplitude from radiance

MijnheerD committed:
Use new utilities function

MijnheerD committed:
Remove unnecessary code

MijnheerD committed:
Add healpy as optional dependency for galactic noise

PhilippLaub committed:
Merge pull request #857 from nu-radio/set-efield-positions

set the correct efield position (relative antenna position) in `readCoREASInterpolator`

MijnheerD committed:
Add noise to each polarisation by not summing

PhilippLaub committed:
Merge branch 'develop' into add-caching-efieldToVoltageConverter

MijnheerD committed:
Update function name

MijnheerD committed:
Rename variable

CamphynR committed:
collect copies into function

CamphynR committed:
add check to evaluate whether a scale parameter directory was passed to begin

CamphynR committed:
cleaner 'and' check

CamphynR committed:
forgot frequencies in add_data_driven_noise

fschlueter committed:
Update channelGenericNoiseAdder.py: change maxsize for caching

PhilippLaub committed:
add docstring, add deprecation warning and use sim_station instead of station

PhilippLaub committed:
correct channel id variable name

PhilippLaub committed:
add proper dosctrings to functions

fschlueter committed:
Remove deprecated config from simulation.py

MijnheerD committed:
Merge pull request #858 from nu-radio/lofar/add-radioshower-info

Expand Radioshower information

martin01021207 committed:
Update stationDeepCRVariables.py

Make the window_size definition consistent

fschlueter committed:
Update geometryUtilities.py: update doc-strings, trim underlining

fschlueter committed:
Update stationDeepCRVariables.py: Some clean up

fschlueter committed:
More cleanup

PhilippLaub committed:
fix typo in doctring

PhilippLaub committed:
add warning, update docstring and reduce code a bit

fschlueter committed:
Update signal_processing.py: Add missing import

sjoerd-bouma committed:
add link to contributing page on documentation to PR template

CamphynR committed:
Merge pull request #854 from nu-radio/add-data-driven-noise

Add data driven noise module that uses scale parameters found by fitting rayleigh distributions per frequency to the data to generate data-informed noise

jakobhenrichs committed:
add fallback solution for the case of inf trigger times

PhilippLaub committed:
make warning more specific

MijnheerD committed:
Add warning for efields below the surface

sjoerd-bouma committed:
fix typo in warning [ci skip]

martin01021207 committed:
Update and rename stationDeepCRVariables.py to stationCoherentlySummedWaveforms.py

Renamed stationDeepCRVariables.py to stationCoherentlySummedWaveforms.py and added doc strings and descriptions.

fschlueter committed:
Update stationCoherentlySummedWaveforms.py: Update doc-string.

martin01021207 committed:
Adding Hit Filter to NuRadioReco

Adding the script of Hit Filter to NuRadioReco

martin01021207 committed:
Update stationHitFilter.py

Added import:
from NuRadioReco.modules.base.module import register_run

fschlueter committed:
Add doc-strings to RNO_G/dataProviderRNOG.py

fschlueter committed:
Add doc-string for has_glitch function

fschlueter committed:
fixing doc-strings

fschlueter committed:
Merge pull request #864 from nu-radio/cleanup-simulation

Remove deprecated `time_resolution` from simulation config

fschlueter committed:
Merge pull request #861 from nu-radio/pr-template

Extend PR template

fschlueter committed:
Merge pull request #808 from nu-radio/SineSub-fs

Implement module for sinewave subtraction

sjoerd-bouma committed:
minor docstring formatting

fschlueter committed:
Remove custom function of block offset removal from mattak reader module and moved it to blockoffset module

fschlueter committed:
Sneak in some clean up

fschlueter committed:
Restructure begin interface

fschlueter committed:
Merge pull request #810 from nu-radio/remove_cable_delay_addition

Remove the automatic addition of cable delays with sim_to_data = False

fschlueter committed:
Moved to functions from trace_utilities.py to signal_processing.py. Add wrapper which return original functions + warning. Changed it everywhere. add doc-string

fschlueter committed:
Move funcion has_glitch from rnog_utilities (removed) to RNO_G/channelGlitchDetector.py

fschlueter committed:
Update dataProviderRNOG.py:fix doc-string

fschlueter committed:
Complete detector description for test

fschlueter committed:
Simplify changes to efieldToVoltageConverter.py. Remove unused variable.

fschlueter committed:
Update reference values for test by 1%. Changes are expected since this PR intorudces changes to the simulated trace lenght which effects the simulated noise/random sequence

fschlueter committed:
Update reference file. This tests runs once without noise and once with. The noiseless test passes which make me believe that the test with noise only fails due to the changes in the trace length which changes the noise

martin01021207 committed:
Update stationHitFilter.py

Fixed typos and made sure the style is following the standard

fschlueter committed:
Merge pull request #862 from nu-radio/add-caching-efieldToVoltageConverter

Add caching to the efieldToVoltageConverter

fschlueter committed:
Merge branch 'develop' into cleanup-utilities

fschlueter committed:
Decide based on efield postion w.r.t. the antenna whether an extra time shift has to be considered in the efieldToVoltageConvert modules

martin01021207 committed:
Update stationHitFilter.py

Added one extra getter function to see the coincidence in PA.

martin01021207 committed:
Update stationHitFilter.py

Made some parts of the code look more clean and concise

martin01021207 committed:
Update stationHitFilter.py

Fixed the wrong indentation in the doc string

fschlueter committed:
Simplify function convert_obs_positions_to_nuradio_on_ground

MijnheerD committed:
Add healpy to list of ALL extras

MijnheerD committed:
Relax in-air condition slightly to avoid warnings for LOFAR

jakobhenrichs committed:
Merge pull request #866 from nu-radio/mattakReader_readout_times

readRNOGDataMattak: add fallback solution for the case of inf trigger times

fschlueter committed:
Merge branch 'develop' into add_rnog_data_processing_script

fschlueter committed:
Merge pull request #790 from nu-radio/add_rnog_data_processing_script

Adding a script for the standard data processing of RNO-G data

martin01021207 committed:
Update stationHitFilter.py

Made some changes and polished the code

martin01021207 committed:
Update stationHitFilter.py

Added back the condition:  sum(is_coincident_in_PA) >= 1

martin01021207 committed:
Update stationHitFilter.py

More code polishing

fschlueter committed:
Merge branch 'develop' into update-rnog-simulation-example

martin01021207 committed:
Merge branch 'develop' into introduce_hit_filter

philippwindischhofer committed:
Merge branch 'develop' into enable_modifications_to_RNOG_detector

philippwindischhofer committed:
make modified detector export properly

philippwindischhofer committed:
fix export of channel positions

fschlueter committed:
Update coreas.py

fschlueter committed:
Merge branch 'develop' into cleanup-utilities

fschlueter committed:
Remove un used import

fschlueter committed:
Remove un used import

philippwindischhofer committed:
export of modified device positions

fschlueter committed:
Make function private

martin01021207 committed:
Hit Filter example script

Added an example script applying the Hit Filter

philippwindischhofer committed:
also convert device id from str back to int

philippwindischhofer committed:
also convert device id from str back to int

fschlueter committed:
Simplifying and optimizing the hitFilter module

fschlueter committed:
Update time checker

fschlueter committed:
Fixes to hit filter

sjoerd-bouma committed:
Merge branch 'develop' of github.com:nu-radio/NuRadioMC into add-efield-galactic-noise

sjoerd-bouma committed:
remove some unused variables from channelGalacticNoiseAdder

sjoerd-bouma committed:
copy in-ice code from channelGalacticNoiseAdder to efieldGalacticNoiseAdder

martin01021207 committed:
Update stationHitFilter.py

Changed back to the original structure for major functions and made some small extra changes.

martin01021207 committed:
Update and rename apply_hit_filter_example.py to apply_hit_filter.py

Exclude FT events and filter only RF events.

fschlueter committed:
Make add_to_trace compatible with 3d traces

fschlueter committed:
Merge branch 'develop' into reader-blockoffset

fschlueter committed:
Fix equation, fix multiple thresholds

Commit:
Add docs for new antenna models

Commit:
Add docs for new antenna models

fschlueter committed:
Update hit filter algorithm

shallmann committed:
use correct LPDA numbering for station 11

MijnheerD committed:
Use vectorised function to avoid list comprehension

sjoerd-bouma committed:
fix typos in docstrings [ci skip]

fschlueter committed:
Update base_trace.py: Following sjoerds suggestion.

shallmann committed:
the additional VPol should not be an HPol

shallmann committed:
fixes LPDA channel positions for station 14

shallmann committed:
fixes theta orientation for all LPDAs

MijnheerD committed:
Merge pull request #860 from nu-radio/add-efield-galactic-noise

Add module for Galactic noise on Efields

shallmann committed:
attempt fix for antenna orientation

shallmann committed:
add some more options to select specific stations / antenna types for plotting

shallmann committed:
Merge branch 'develop' into fix/correct_station14_station11_geometries

martin01021207 committed:
Update stationHitFilter.py

Some changes to Felix's modified and optimized Hit Filter code

martin01021207 committed:
Merge branch 'introduce_hit_filter' into introduce_hit_filter-fs

martin01021207 committed:
Merge pull request #874 from nu-radio/introduce_hit_filter-fs

Introduce hit filter fs

martin01021207 committed:
Delete NuRadioReco/examples/RNOG/apply_hit_filter.py

Delete apply_hit_filter.py and add the Hit Filter application to standard RNO-G data analysis example scripts instead.

martin01021207 committed:
Update data_analysis_example.py

Added Hit Filter to this example.

martin01021207 committed:
Update data_analysis_example_advanced.py

Added Hit Filter application to this example.

martin01021207 committed:
Update data_analysis_example.py

Added one line

martin01021207 committed:
Update data_analysis_example_advanced.py

Added one line.

martin01021207 committed:
Update stationCoherentlySummedWaveforms.py

Adding other CSW variables

martin01021207 committed:
Update parameters.py

Add new CSW vars to parameters

martin01021207 committed:
Update stationCoherentlySummedWaveforms.py

Fixed typo

fschlueter committed:
Merge pull request #875 from nu-radio/extend-bt

Make add_to_trace compatible with 3d traces

fschlueter committed:
Merge pull request #868 from nu-radio/reader-blockoffset

Move implementation of median-blockoffset-removal function from reader to dedicated module

shallmann committed:
adjust downward facing depth (1.4m according to datasheet matching rough measurement in the office)

fschlueter committed:
Merge pull request #871 from nu-radio/simplify-coreas

Simplify function convert_obs_positions_to_nuradio_on_ground

martin01021207 committed:
Merge pull request #877 from nu-radio/adding_other_csw_variables

Add other CSW variables in stationCoherentlySummedWaveforms.py

martin01021207 committed:
Update data_analysis_example.py

Now also applying the Hit Filter to FT events too

martin01021207 committed:
Update data_analysis_example_advanced.py

Now also applying the Hit Filter to FT events too

martin01021207 committed:
Update data_analysis_example.py

Typo

martin01021207 committed:
Update data_analysis_example_advanced.py

Typo

shallmann committed:
adjust also other station 11 LPDA numberings

shallmann committed:
Merge branch 'fix/correct_station14_station11_geometries' of github.com:nu-radio/NuRadioMC into fix/correct_station14_station11_geometries

shallmann committed:
Merge branch 'develop' into fix/correct_station14_station11_geometries

martin01021207 committed:
Update stationHitFilter.py

Added logger

martin01021207 committed:
Update data_analysis_example.py

Change after logger added in Hit Filter

martin01021207 committed:
Update data_analysis_example_advanced.py

Change after logger added in Hit Filter

shallmann committed:
Merge pull request #876 from nu-radio/fix/correct_station14_station11_geometries

Fix/correct station14 station11 geometries

fschlueter committed:
Merge branch 'develop' into cleanup-utilities

fschlueter committed:
Change to DeprecationWarning, add whitespaces

fschlueter committed:
Moved a few more utilities functions around

fschlueter committed:
Move another function - no need to add a wrapper at old location

fschlueter committed:
Merge branch 'cleanup-utilities' into improve-detector-simulation

fschlueter committed:
Missed that

fschlueter committed:
Missed that

fschlueter committed:
Add warning

fschlueter committed:
Make code a bit more modular - move resampling function to utilities and use it in framework class. Only log warning if relevant

fschlueter committed:
Add doc-string [ci-skip]

fschlueter committed:
Update signal_processing.py: Fix imports

fschlueter committed:
Move around add_cable_delay function to avoid circular import

sjoerd-bouma committed:
add hashes for new RNO-G antennas

sjoerd-bouma committed:
add hashes for new RNO-G antennas

fschlueter committed:
Add missing import

sjoerd-bouma committed:
Change LOFAR reader modules to use NuRadio units everywhere

martin01021207 committed:
Update stationHitFilter.py

Discarded the multi-threshold feature and made some changes in the logging info.

martin01021207 committed:
Update data_analysis_example.py

Small changes in the Hit Filter part

martin01021207 committed:
Update data_analysis_example_advanced.py

Some small changes in the Hit Filter part

martin01021207 committed:
Update stationHitFilter.py

Small fix in the getter function:  is_in_time_window_PA()

sjoerd-bouma committed:
fix formatting of logger debug output in antennapattern

sjoerd-bouma committed:
improve/fix docstrings in io/LOFAR

fschlueter committed:
Cleanup and improving doc-strings

fschlueter committed:
Add argument to specify trigger type. Add time logging

fschlueter committed:
Clean up example script.

fschlueter committed:
Update signal_processing.py: Fix imports

fschlueter committed:
Use proper units and add logger

fschlueter committed:
Do not try to load run table by default

fschlueter committed:
Do not try to load run table by default

sjoerd-bouma committed:
fix typo in variable [ci skip]

fschlueter committed:
Update signal_processing.py: Fix unit names

sjoerd-bouma committed:
Revert "fix typo in variable [ci skip]"

This reverts commit 6fd807143c1e796d34690cb5b15f43a7480549fd.

sjoerd-bouma committed:
Revert "Change LOFAR reader modules to use NuRadio units everywhere"

This reverts commit 550f2d3a9ef4fb53f75622687646190fb5f33ec9.

martin01021207 committed:
Update stationHitFilter.py

Small changes in the comments

martin01021207 committed:
Merge branch 'develop' into introduce_hit_filter

fschlueter committed:
Merge pull request #881 from nu-radio/mattak-reader

Do not try to load run table by default

fschlueter committed:
Merge pull request #873 from nu-radio/antenna_update

Add docs for new antenna models

fschlueter committed:
Drop explicit units.Hz^-1 from equations after converting impedence and boltzman constant into nuradio unit system.

fschlueter committed:
Fix typo [ci skip]

fschlueter committed:
Add doc-strings to modules

fschlueter committed:
contine removing is_cosmic_ray

sjoerd-bouma committed:
change all outputs of readLOFARData to Nuradio units

sjoerd-bouma committed:
make rawTBBio modules private

sjoerd-bouma committed:
apply LOFAR calibration delays inside readLOFARData

sjoerd-bouma committed:
allow stationRFIFilter to use _rawTBBio... for now

cg-laser committed:
add documentation

fschlueter committed:
Move module bandpass_filter.py to _deprecated folder

martin01021207 committed:
Merge pull request #867 from nu-radio/introduce_hit_filter

Adding Hit Filter to NuRadioReco

fschlueter committed:
Merge branch 'cleanup-utilities' into patch-fs

fschlueter committed:
Merge pull request #878 from nu-radio/patch-fs

Implement core resampling method (from BaseTrace class) in utilities. Make warning smarter

fschlueter committed:
Move implementation from get_filter_response to signal_processing.py - add deprecation warning - changing it everywhere in NuRadio

fschlueter committed:
Add doc-string [ci skip]

jakobhenrichs committed:
move variableWindowSizeCorrelation to trace_utilities

fschlueter committed:
Strip a few more things from variableWindowSizeCorrelation.py

fschlueter committed:
Move variableWindowSizeCorrelation.py to _deprecated/

fschlueter committed:
Refactoring get_variable_window_size_correlation

fschlueter committed:
Fix importing deprecated modules

fschlueter committed:
Forgot to commit this file

philippwindischhofer committed:
extend docstring

fschlueter committed:
Cleanup

fschlueter committed:
Fix interferometry.get_time_shifts_plane

fschlueter committed:
Add doc-string to utilties package. [ci skip]

philippwindischhofer committed:
add ability to add comments to exported detector description

fschlueter committed:
Merge pull request #844 from nu-radio/update-rnog-simulation-example

Update rnog simulation example

philippwindischhofer committed:
docstring syntax

sjoerd-bouma committed:
Merge pull request #879 from nu-radio/hotfix/antennapattern-debug-formatting

Fix formatting of logger debug output in antennapattern

sjoerd-bouma committed:
vectorize get_time_delay_from_direction

fschlueter committed:
Remove implementation from get_time_shifts_plane and return get_time_delay_from_direction + warning

sjoerd-bouma committed:
support position arrays with ndim > 2

sjoerd-bouma committed:
Merge branch 'master' of github.com:nu-radio/NuRadioMC into rc-v3.0.0

fschlueter committed:
Merge branch 'develop' into cleanup-utilities + Changes to the
_get_number_of_samples function

fschlueter committed:
WS [ci skip]

sjoerd-bouma committed:
update version number

sjoerd-bouma commented:
@fschlueter my suggestion would be to merge #880 and #890 (+ whatever still happens to get merged today) and then release?

sjoerd-bouma committed:
Merge pull request #880 from nu-radio/lofar/unit-fixes

Use NuRadio units in io/LOFAR

fschlueter commented:
> @fschlueter my suggestion would be to merge #880 and #890 (+ whatever still happens to get merged today) and then release?

Sounds good.

fschlueter committed:
Remove module NuRadioReco/modules/efieldAirToIcePropagator.py as it missed the actual implementation

fschlueter committed:
Move traceWindows.py -> _deprecated/traceWindows.py

fschlueter committed:
Move implementation of get_window_around_maximum to NuRadioReco.utilities.diodeSimulator.get_window_around_maximum

fschlueter committed:
Update changelog.txt

fschlueter committed:
Using the efield and antenna position to decide whether additional propagation is necessary. Add doc strings

fschlueter committed:
Merge branch 'develop' into improve-detector-simulation

fschlueter committed:
Reimplement function calculate_vrms_from_temperature which apparently got lost when resolving some merge conflict. Expand module doc-string

fschlueter committed:
Merge pull request #889 from nu-radio/rnog_detector_improvements

Small fix and improvement to RNO-G detector class

fschlueter committed:
Merge pull request #888 from nu-radio/interferometry-utility-patch

Fix interferometry.get_time_shifts_plane

fschlueter committed:
Merge pull request #886 from nu-radio/docstrings

Docstring

fschlueter committed:
Fix documentation syntax

fschlueter committed:
Make Sjoerd happy

fschlueter committed:
Merge pull request #891 from nu-radio/remove-module

Remove module NuRadioReco/modules/efieldAirToIcePropagator.py

fschlueter committed:
Merge pull request #869 from nu-radio/cleanup-utilities

Move functions around in NuRadioReco/utilities

fschlueter committed:
Merge branch 'develop' into improve-detector-simulation

fschlueter committed:
Merge pull request #686 from nu-radio/enable_modifications_to_RNOG_detector

Enable modifications of the RNO-G detector description

fschlueter committed:
Make merge_hdf5.py more robust: Replaced np.all(a == b) with np.allclose(a, b)

fschlueter committed:
Avoid divison through 0 in the channelSignalReconstructor. Disable VEL caching in simulation.py. Add better warning to efieldToVoltageConverter

fschlueter committed:
Merge pull request #893 from nu-radio/fix-logging-and-avoid-numpy-warning

Avoid zero-division in channelSignalReconstructor. Disable VEL caching in simulation.py

fschlueter committed:
Merge pull request #870 from nu-radio/improve-detector-simulation

Remove artificial dependency on "is_cosmic_ray" in efieldToVoltageConverter modules

fschlueter committed:
Optimize logging in simulation.py

fschlueter committed:
Obey flake8 rule F824

fschlueter committed:
Merge pull request #900 from nu-radio/hotfix-flake8

Obey flake8 rule F824

fschlueter committed:
Revert "Add data driven noise"

sjoerd-bouma committed:
Merge branch 'develop' into rc-v3.0.0

fschlueter committed:
Merge pull request #902 from nu-radio/revert-854-add-data-driven-noise

Revert "Add data driven noise"

fschlueter committed:
Merge pull request #892 from nu-radio/improve-merge_hdf5

Make merge_hdf5.py more robust

fschlueter committed:
Merge pull request #898 from nu-radio/optimize-logging-simulations

Optimize logging in simulation.py

fschlueter committed:
Merge pull request #890 from nu-radio/rc-v3.0.0

Update version number for release v3.0.0

sjoerd-bouma commented:
@fschlueter From my side I'm happy to release, but I can't approve  - feel free to approve (and merge). Not sure why proposal failed to install on python 3.x, I ran it again manually (https://github.com/nu-radio/NuRadioMC/actions/runs/14386670421/job/40343684733) and it seems to install fine there - looks like maybe a temporary server failure for the bzip2 installation.
